plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSessionProperties.java

```diff
@@ -677,6 +710,12 @@ public static boolean isQueryPartitionFilterRequired(ConnectorSession session)
         return session.getProperty(QUERY_PARTITION_FILTER_REQUIRED, Boolean.class);
     }
 
+    @SuppressWarnings("unchecked cast")
+    public static Set<String> getQueryPartitionFilterRequiredSchemas(ConnectorSession session)
+    {
+        return (Set<String>) session.getProperty(QUERY_PARTITION_FILTER_REQUIRED_SCHEMAS, Set.class);
```

Comment:
`Optional.ofNullable`

Reply:
Instead of using Optional, we could try this approach right ? If the new list contains schemas, then we only check those schemas else we check for all schemas.

Reply:
you're right. you can add `rnn` check here.